cpp/src/arrow/compute/api_scalar.h

```diff
@@ -108,6 +108,32 @@ class ARROW_EXPORT RoundToMultipleOptions : public FunctionOptions {
   RoundMode round_mode;
 };
 
+class ARROW_EXPORT RoundTemporalOptions : public FunctionOptions {
+ public:
+  enum Ambiguous { AMBIGUOUS_RAISE, AMBIGUOUS_EARLIEST, AMBIGUOUS_LATEST };
+  enum Nonexistent { NONEXISTENT_RAISE, NONEXISTENT_EARLIEST, NONEXISTENT_LATEST };
+
+  explicit RoundTemporalOptions(std::string frequency = "day",
```

Comment:
I realize this is just a draft :) that said, should be an enum as well?

Reply:
Too soon David :)

Reply:
Thanks for the feedback though! will do.